xdsl/irdl/declarative_assembly_format_parser.py

```diff
@@ -371,15 +373,35 @@ def parse_optional_variable(
                 # Always qualify builtin attributes
                 # This is technically an approximation, but appears to be good enough
                 # for xDSL right now.
-                if unique_base is not None and unique_base in Builtin.attributes:
+                unique_type = None
+                if unique_base is not None and issubclass(unique_base, TypedAttribute):
+                    constr = attr_def.constr
+                    # TODO: generalize.
```

Comment:
Can you please expand on this comment, and file an issue?

Reply:
https://github.com/xdslproject/xdsl/Done: issues/2499